app/js/components/manageApps/ManageApps.jsx

```diff
@@ -106,48 +108,61 @@ export default class ManageApps extends React.Component {
       let new_zip = new JSZip();
       new_zip.loadAsync(zipedAddon)
         .then((zip) => {
-          return zip.files['manifest.webapp'];
+          if (zip.files['manifest.webapp']) {
+            return zip.files['manifest.webapp'];
+          }
+          return false;
         })
         .then((file) => {
+          if(!file) return false;
           return new_zip.file(file.name).async("string");
         })
         .then(result => {
-          resolve (JSON.parse(result));
-        });
+          resolve(JSON.parse(result));
+        })
+        .catch(e => resolve(false));
```

Comment:
Any plans to handle this error in the future?

Reply:
I am already handling it by resolving it to false, that way the next Promise `readZippedOn` when chained with `then` can handle it as a `falsy` file. Hence, not an `OWA` module.